### PR TITLE
Feature/add managedlocations support

### PR DESCRIPTION
**Pull request by SCBbestof**

### CODE REVIEW

File: smartlogic/service.go [hunk 1]
```diff
@@ -237,12 +248,24 @@ func appendFactsetConcordances(concordances []ConcordedId, concept Concept, conc
 
 func appendLocationConcordances(concordances []ConcordedId, conceptIdentifiers []LocationType, conceptUuid string, authority string, tid string) ([]ConcordedId, error) {
 	for _, id := range conceptIdentifiers {
+		if len(strings.TrimSpace(id.Value)) == 0 {
+			err := errors.New("Bad Request: Payload from Smartlogic contains one or more empty " + authority + " values")
+			log.WithFields(log.Fields{"transaction_id": tid, "UUID": conceptUuid}).Error(err)
+			return nil, err
```
Comment (tosan88) [inline smartlogic/service.go]: IIRC the expected behavior is to let the concordance work for the valid values, and only log empty values as warning

File: smartlogic/service.go [hunk 2]
```diff
@@ -237,12 +248,24 @@ func appendFactsetConcordances(concordances []ConcordedId, concept Concept, conc
...
 		uuidFromConceptIdentifier := convertToUuid(id.Value)
 		if conceptUuid == uuidFromConceptIdentifier {
 			err := errors.New("Bad Request: Payload from smartlogic has a smartlogic uuid that is the same as the uuid generated from " + authority + " id")
 			log.WithFields(log.Fields{"transaction_id": tid, "UUID": conceptUuid}).Error(err)
 			return nil, err
 		}
+		if concordancesContainValue(concordances, uuidFromConceptIdentifier) {
+			err := errors.New("Bad Request: Payload from Smartlogic contains duplicate " + authority + " values")
+			log.WithFields(log.Fields{"transaction_id": tid, "UUID": conceptUuid}).Error(err)
+			return nil, err
```
Comment (tosan88) [inline smartlogic/service.go]: Same here: the expected behavior is to let the concordance work for the valid values, and only log duplicate values as warning

File: smartlogic/model.go [hunk 3]
```diff
@@ -1,33 +1,39 @@
...
-	TmeIdentifiersEditorial           []TmeId        `json:"http://www.ft.com/ontology/TMEIdentifier,omitempty"`
-	TmeIdentifiersManagedLocation     []TmeId        `json:"http://www.ft.com/ontology/managedlocation/TMEIdentifier,omitempty"`
-	FactsetIdentifiersEditorial       []FactsetId    `json:"http://www.ft.com/ontology/factsetIdentifier,omitempty"`
-	FactsetIdentifiersManagedLocation []FactsetId    `json:"http://www.ft.com/ontology/managedlocation/factsetIdentifier,omitempty"`
-	DbpediaIdentifiers                []LocationType `json:"http://www.ft.com/ontology/managedlocation/dbpediaId,omitempty"`
-	GeonamesIdentifiers               []LocationType `json:"http://www.ft.com/ontology/managedlocation/geonamesId,omitempty"`
-	WikidataIdentifiers               []LocationType `json:"http://www.ft.com/ontology/managedlocation/wikidataId,omitempty"`
+	ID             string   `json:"@id"`
+	Types          []string `json:"@type,omitempty"`
+	currentConcept Concepter
```
Comment (SCBbestof) [inline smartlogic/model.go]: Why not just *Concepter ?

**Comment by SCBbestof** [comment]
I can't approve since it's my PR, but it looks awesome from my side too 👍 😅

**Pull request merged**